.github/workflows/modules/Steps.pkl

```diff
@@ -0,0 +1,40 @@
+import "package://pkg.pkl-lang.org/github.com/stefma/pkl-gha/com.github.action@0.0.3#/GitHubAction.pkl"
+
+function checkout(customWith: Mapping<String, String|Number|Boolean>?): GitHubAction.Step = new {
+  name = "Checkout"
+  uses = "actions/checkout@v4"
+  with = customWith
+}
```

Comment:
Similarly here, a function isn't necessary for this in Pkl (actually, functions are *usually* not necessary in Pkl!).

You can implement it like:
```suggestion
checkout: GitHubAction.Step = new {
  name = "Checkout"
  uses = "actions/checkout@v4"
}
```

And then a consumer can use it and amend the `with` value like:
```
steps {
  (checkout) {
    with = "my custom with"
  }
}
```

Reply:
Done here https://github.com/pkl-community/setup-pkl/pull/47/commits/e3b40c36d8f4f654ce498dd774c616c01fe4dc29